signalfx-java/src/main/java/com/signalfx/signalflow/WebSocketTransport.java

```diff
@@ -0,0 +1,592 @@
+        }
+    }
+
+    @Override
+    public Channel attach(String handle, final Map<String, String> parameters) {
+        if (log.isDebugEnabled()) {
+            log.debug("attach: [ " + handle + " ] with parameters: " + parameters);
+        }
+
+        Channel channel = new TransportChannel(transportConnection);
+        HashMap<String, String> request = new HashMap<String, String>(parameters);
+        request.putAll(buildParams("type", "attach", "handle", handle));
+        transportConnection.sendMessage(channel, request);
+
+        return channel;
+    }
+
+    @Override
+    public Channel execute(String program, final Map<String, String> parameters) {
+        if (log.isDebugEnabled()) {
+            log.debug("execute: [ " + program + " ] with parameters: " + parameters);
+        }
+
+        Channel channel = new TransportChannel(transportConnection);
+        HashMap<String, String> request = new HashMap<String, String>(parameters);
+        request.putAll(buildParams("type", "execute", "program", program));
+        transportConnection.sendMessage(channel, request);
+
+        return channel;
+    }
+
+    @Override
+    public void start(String program, final Map<String, String> parameters) {
+        if (log.isDebugEnabled()) {
+            log.debug("start: [ " + program + " ] with parameters: " + parameters);
+        }
+
+        HashMap<String, String> request = new HashMap<String, String>(parameters);
+        request.putAll(buildParams("type", "start", "program", program));
+
+        transportConnection.sendMessage(request);
+    }
+
+    @Override
+    public void stop(String handle, final Map<String, String> parameters) {
+        if (log.isDebugEnabled()) {
+            log.debug("stop: [ " + handle + " ] with parameters: " + parameters);
+        }
+
+        HashMap<String, String> request = new HashMap<String, String>(parameters);
+        request.putAll(buildParams("type", "stop", "handle", handle));
+
+        transportConnection.sendMessage(request);
+    }
+
+    @Override
+    public void close(int code, String reason) {
+        if ((transportConnection.getConnection() != null)
+                && (transportConnection.getConnection().isOpen())) {
+            transportConnection.close(code, reason);
+            try {
+                this.webSocketClient.getFactory().stop();
+            } catch (Exception ex) {
+                log.error("error while stopping websocketfactory", ex);
+            }
+            log.info("transport closed");
+        }
+    }
+
+    @Override
+    public void keepalive(String handle) {
+        if (log.isDebugEnabled()) {
+            log.debug("keepalive: [ " + handle + " ]");
+        }
+
+        Map<String, String> request = buildParams("type", "keepalive", "handle", handle);
+        transportConnection.sendMessage(request);
+    }
+
+    private static Map<String, String> buildParams(Object... params) {
```

Comment:
If you still use this here given my comment above, don't have it twice between the Websocket transport and the SSE transport, but that in some shared class.